src/components/Form/Form.tsx

```diff
@@ -0,0 +1,124 @@
+import "./style.scss";
+import { useFormik } from "formik";
+import { Film, Modal } from "../../interfaces";
+import { films } from "../../films";
+import Select from "react-select";
+import DeleteForm from "./DeleteForm";
+import { useEffect, useState } from "react";
+import { newMovie, FormType, Genres } from "../../constants";
+import { selectStyle } from "./selectConfig";
+
+interface FormProps {
+  closeModal(): void;
+  modalState: Modal;
+}
+
+const Form: React.FC<FormProps> = ({ closeModal, modalState }) => {
+  const [movie] = useState<Film>(
+    films.find((el) => {
+      return el.id === modalState.filmID;
+    }) || newMovie
+  );
+
+  useEffect(() => {
+    document.querySelector("body")?.classList.add("overflow-hidden");
+    return () => {
+      document.querySelector("body")?.classList.remove("overflow-hidden");
+    };
+  });
+
+  const formik = useFormik({
+    initialValues: movie,
+    onSubmit: (values) => {
+      values.releaseDate = new Date(values.releaseDate);
+      if (modalState.type === FormType.EDIT) {
+        films[films.indexOf(movie)] = values;
+      } else {
+        films.push(values);
+      }
+      closeModal();
+    },
+    onReset: (values) => {
+      values = movie;
+    },
+  });
+
+  return (
+    <div className="form-wrapper">
+      {modalState.type === FormType.DELETE ? (
+        <DeleteForm movie={movie} closeModal={closeModal} />
+      ) : (
+        <form
```

Comment:
think about another way to handle this. What if tomorrow you need to add another bunch of modals.